Sync/KeyBundle.swift

```diff
@@ -172,11 +172,29 @@ open class KeyBundle: Hashable {
     open func serializer<T: CleartextPayloadJSON>(_ f: @escaping (T) -> JSON) -> (Record<T>) -> JSON? {
         return { (record: Record<T>) -> JSON? in
             let json = f(record.payload)
-            let data = (json.rawString([writtingOptionsKeys.castNilToNSNull: true]) ?? "").utf8EncodedData
+            if json.isNull() {
```

Comment:
Would it make sense to move this to an extension on SwiftyJSON so we can use it in other places we `.utf8Encoded` SwiftJSON values? I see we also do this in `FxAClient10.swift` and `PushClient.swift`.